packages/astro/src/runtime/server/index.ts

```diff
@@ -296,11 +296,17 @@ function createFetchContentFn(url: URL, site: URL) {
 				}
 				const urlSpec = new URL(spec, url).pathname;
 				return {
-					...mod.frontmatter,
-					Content: mod.default,
-					content: mod.metadata,
 					file: new URL(spec, url),
 					url: urlSpec.includes('/pages/') ? urlSpec.replace(/^.*\/pages\//, sitePathname).replace(/(\/index)?\.md$/, '') : undefined,
+					data: mod.frontmatter,
+					getContent: () => mod.default().then((m: any) => m.default),
```

Comment:
This probably doesn't need to be updated to have an explicit getter function! 

Astro should unwrap any promises before attempting to render them, so we could just keep this name as `Content` and it would still be deferred until it is rendered.